### PR TITLE
feat: added support for fastify v5

### PR DESCRIPTION
**Pull request by instanacd**
Tada!

refs https://jsw.ibm.com/browse/INSTA-15011

### CODE REVIEW

**Comment by kirrg001** [comment]
https://github.com/fastify/fastify/releases/tag/v5.0.0

**Comment by kirrg001** [comment]
```

Fastify (533):	startup failure TypeError: Cannot create property 'host' on string '4329'
@instana/collector:     at Object.listen (/artifacts/node_modules/fastify/lib/server.js:60:26)
@instana/collector:     at start (/artifacts/packages/collector/test/tracing/frameworks/fastify/app.js:136:15)
@instana/collector:     at Object.<anonymous> (/artifacts/packages/collector/test/tracing/frameworks/fastify/app.js:144:1)
@instana/collector:     at Module._compile (node:internal/modules/cjs/loader:1469:14)
@instana/collector:     at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
@instana/collector:     at Module.load (node:internal/modules/cjs/loader:1288:32)
@instana/collector:     at Module._load (node:internal/modules/cjs/loader:1104:12)
@instana/collector:     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
@instana/collector:     at node:internal/main/run_main_module:28:49

```

**Comment by kirrg001** [comment]
I have identified:

> feat: emit diagnostics_channel events upon routing request by @tlhunter in https://github.com/fastify/fastify/pull/5252

Extended https://jsw.ibm.com/browse/INSTA-804

**Comment by kirrg001** [comment]
v5 no longer supports Node < 20. https://github.com/fastify/fastify/pull/5481/files
It throws an error with diagnostic channel.

> TypeError: diagnostics.tracingChannel is not a function

**Comment by kirrg001** [comment]
`app.listen` changed. https://github.com/fastify/fastify/pull/4900

**Comment by kirrg001** [comment]
@aryamohanan @abhilash-sivan RFR

**Comment by aryamohanan** [comment]
Useful guide: [Fastify Migration Guide V5](https://fastify.dev/docs/latest/Guides/Migration-Guide-V5/).

File: packages/core/src/tracing/instrumentation/frameworks/fastify.js [hunk 1]
```diff
@@ -59,8 +59,16 @@ function instrument(build) {
     app.addHook('onRequest', function onRequest(request, reply, done) {
       try {
         // NOTE: v1 uses _context https://github.com/fastify/fastify/blob/1.x/fastify.js#L276
-        //       v2/v3 uses context https://github.com/fastify/fastify/blob/2.x/test/handler-context.test.js#L41
-        const url = reply._context ? reply._context.config.url : reply.context.config.url;
+        //       v2/v3/v4 uses context https://github.com/fastify/fastify/blob/2.x/test/handler-context.test.js#L41
+        //       v5 uses reply.routeOptions and no longer exposes `.context`
```
Comment (kirrg001) [inline packages/core/src/tracing/instrumentation/frameworks/fastify.js:63]: Thanks @aryamohanan for catching this.
There was an easier way of accessing the original url. See https://fastify.dev/docs/latest/Guides/Migration-Guide-V5/#streamlined-access-to-route-definition

**Review by aryamohanan — APPROVED** [review]
LGTM

**Pull request merged**